tools/src/commands/PublishProjectTemplates.ts

```diff
@@ -51,6 +37,45 @@ async function action(options) {
     options.sdkVersion = sdkVersion;
   }
 
+  const sdkTag = `sdk-${semver.major(options.sdkVersion)}`;
+
+  const tagOptions = new Map<string, string[]>();
+  if (semver.prerelease(options.sdkVersion)) {
+    tagOptions.set(`${sdkTag} and next`, [sdkTag, 'next']);
+  } else {
+    tagOptions.set(`${sdkTag} and latest`, [sdkTag, 'latest']);
+  }
+  tagOptions.set(`${sdkTag} and beta`, [sdkTag, 'beta']);
+  tagOptions.set(sdkTag, [sdkTag]);
+
+  const { tagChoice } = await inquirer.prompt<{ tagChoice: string }>([
+    {
+      type: 'list',
+      name: 'tagChoice',
+      prefix: '❔',
+      message: 'Which tags would you like to use?',
+      choices: [...tagOptions.keys(), 'custom'],
+    },
+  ]);
+
+  let tags = tagOptions.get(tagChoice) || [sdkTag];
+
+  if (tagChoice === 'custom') {
+    const { customTag } = await inquirer.prompt<{ customTag: string }>([
+      {
+        type: 'input',
+        name: 'customTag',
+        message: 'Enter custom tag string:',
+        default: 'custom',
+      },
+    ]);
+    tags = [customTag];
+  }
```

Comment:
small style nit - i'd probably re-arrange like this:

```js
  let tags = [];

  if (tagChoice === 'custom') {
    const { customTag } = await inquirer.prompt<{ customTag: string }>([
      {
        type: 'input',
        name: 'customTag',
        message: 'Enter custom tag string:',
        default: 'custom',
      },
    ]);
    tags = [customTag];
  } else {
    tags = tagOptions.get(tagChoice);
  }
```

or like this

```js
  let tags = tagChoice === 'custom' ? await promptForCustomTagAsync() : tagOptions.get(tagChoice);
```

where `promptForCustomTagAsync()` is extracted to a helper

Reply:
Fixed